### PR TITLE
Clean up factories

### PR DESCRIPTION
**Pull request by lowjoel**

### CODE REVIEW

File: app/views/course/achievements/_form.html.slim [hunk 1]
```diff
@@ -27,7 +27,7 @@
           th = t('course.condition.type.header')
           th = t('course.condition.description.header')
       tbody
-        - @achievement.conditions.includes(:conditional).each do |condition|
+        - @achievement.conditions.each do |condition|
```
Comment (lowjoel) [inline app/views/course/achievements/_form.html.slim]: @dariusf why was this eager loading specified?

Comment (dariusf) [inline app/views/course/achievements/_form.html.slim]: Probably something I forgot to revert after removing what was using it. My bad.

**Comment by lowjoel** [comment]
@allenwq I think this should be safe to merge even if Coveralls is down.

**Comment by allenwq** [comment]
:+1:

File: spec/factories/course_assessment_assessments.rb [hunk 2]
```diff
@@ -1,20 +1,8 @@
...
-      options[:course] = course if course
-      build(:course_lesson_plan_item, options)
-    end
-    tab do
-      options = {}
-      options[:course] = course if course
-      build(:course_assessment_tab, options)
-    end
+  factory :course_assessment_assessment, class: Course::Assessment, aliases: [:assessment],
+          parent: :course_lesson_plan_item do
```
Comment (lowjoel) [inline spec/factories/course_assessment_assessments.rb]: good job, Hound, for commenting many hours after this was merged =.=

File: app/views/course/achievements/_form.html.slim (hunk 1 above)
Comment (dariusf) [inline app/views/course/achievements/_form.html.slim]: So it turns out this is needed, as `conditional` is used in the partials that are rendered in the block. I've reverted this in #425. Should I assign `@conditions` in the controller to eagerly load this?

**Pull request merged**